File: C++/初级/C++ 编程思想/C/java/chap8a/Employee.java
public class Employee {
    private String last;
    private String first;
    private String title;
    private int age;
    
    public Employee(String l, String f,
                    String t, int a) {
        last = l;
        first = f;
        title = t;
        age = a;
    }
    public String toString() {
        return "{" + last + "," + first +
               "," + title+ "," + age + "}";
    }
    public void setLast(String s) {
        last = s;
    }
    public void setFirst(String s) {
        first = s;
    }
    public void setTitle(String s) {
        title = s;
    }
    public void setAge(int n) {
        age = n;
    }
    public static void main(String[] args) {
        Employee e = new Employee("Jordan", "Michael",
                                  "Guard", 37);
        System.out.println(e);
        e.setLast("Malone");
        e.setFirst("Karl");
        e.setTitle("Center");
        e.setAge(36);
        System.out.println(e);
    }
}

File: C++/初级/C++ 编程思想/C/cplus/chap9b/solution/employee.cpp
// employee.cpp
#include "employee.h"
#include <iostream>
#include <string.h>

Employee::Employee(const char* lst, const char* frst, const char* ttle, int slry) {
    strcpy(last, lst);
    strcpy(first, frst);
    strcpy(title, ttle);
    salary = slry;
}

const char* Employee::getLast() const {
    return last;
}

const char* Employee::getFirst() const {
    return first;
}

const char* Employee::getTitle() const {
    return title;
}

int Employee::getSalary() const {
    return salary;
}

void Employee::setLast(const char* lst) {
    strcpy(last, lst);
}

void Employee::setFirst(const char* frst) {
    strcpy(first, frst);
}

void Employee::setTitle(const char* ttle) {
    strcpy(title, ttle);
}

void Employee::setSalary(int slry) {
    salary = slry;
}

void Employee::print() const {
    std::cout << '{' << last << ',' << first
              << ',' << title << ',' << salary
              << '}';
}
